cpp/src/arrow/compute/kernels/scalar_compare.cc

```diff
@@ -439,6 +472,330 @@ struct ScalarMinMax {
+    }
+    return Status::OK();
+  }
+
+  static Status ExecContainingArrays(KernelContext* ctx,
+                                     const ElementWiseAggregateOptions& options,
+                                     const ExecBatch& batch, Datum* out) {
+    const auto batch_type = batch[0].type();
+    const auto binary_type = checked_cast<const FixedSizeBinaryType*>(batch_type.get());
+    int32_t byte_width = binary_type->byte_width();
+    // Presize data to avoid reallocations
+    int64_t final_size = 0;
+    for (int64_t i = 0; i < batch.length; i++) {
+      auto size = CalculateRowSize(options, batch, i, byte_width);
+      if (size > 0) final_size += size;
+    }
+    FixedSizeBinaryBuilder builder(batch_type);
+    RETURN_NOT_OK(builder.Reserve(batch.length));
+    RETURN_NOT_OK(builder.ReserveData(final_size));
+
+    std::vector<string_view> valid_cols(batch.values.size());
+    for (size_t row = 0; row < static_cast<size_t>(batch.length); row++) {
+      size_t num_valid = 0;
+      for (size_t col = 0; col < batch.values.size(); col++) {
+        if (batch[col].is_scalar()) {
+          const auto& scalar = *batch[col].scalar();
+          if (scalar.is_valid) {
+            valid_cols[col] = UnboxScalar<FixedSizeBinaryType>::Unbox(scalar);
+            num_valid++;
+          } else {
+            valid_cols[col] = string_view();
+          }
+        } else {
+          const auto& array = *batch[col].array();
+          if (!array.MayHaveNulls() ||
+              bit_util::GetBit(array.buffers[0]->data(), array.offset + row)) {
+            const auto data = array.GetValues<uint8_t>(1, /*absolute_offset=*/0);
+            valid_cols[col] = string_view(
+                reinterpret_cast<const char*>(data) + row * byte_width, byte_width);
+            num_valid++;
+          } else {
+            valid_cols[col] = string_view();
+          }
+        }
+      }
+
+      if (num_valid < batch.values.size() && !options.skip_nulls) {
+        // We had some nulls
+        builder.UnsafeAppendNull();
+        continue;
+      }
+      auto result = valid_cols.front();
+      for (size_t col = 1; col < batch.values.size(); ++col) {
+        const auto value = valid_cols[col];
+        if (value.empty()) {
+          DCHECK(options.skip_nulls);
+          continue;
+        }
+        result = result.empty() ? value : Op::Call(result, value);
+      }
+      if (result.empty()) {
+        builder.UnsafeAppendNull();
+      } else {
+        builder.UnsafeAppend(result);
+      }
+    }
+
+    std::shared_ptr<Array> string_array;
+    RETURN_NOT_OK(builder.Finish(&string_array));
+    *out = *string_array->data();
+    out->mutable_array()->type = batch[0].type();
+    DCHECK_EQ(batch.length, out->array()->length);
+    return Status::OK();
+  }
+
+  // Compute the length of the output for the given position, or -1 if it would be null.
+  static int64_t CalculateRowSize(const ElementWiseAggregateOptions& options,
```

Comment:
Why do you need this function? The input types should all have the same width (it is ensured by `ResolveMinOrMaxOutputType`).

Reply:
Removed it, it was overkill - it was intended to handle the null case, but I don't think that even matters at all for fixed size binary.